components/builder-api/src/server/resources/channels.rs

```diff
@@ -678,9 +680,67 @@ fn get_package_fully_qualified(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn get_latest_packages_for_origin_channel(req: HttpRequest,
+                                          path: Path<(String, String)>,
+                                          qtarget: Query<Target>)
+                                          -> HttpResponse {
+    let (origin, channel) = path.into_inner();
+    let channel = ChannelIdent::from(channel);
+
+    match do_get_latest_channel_packages(&req, &qtarget, &origin, &channel) {
+        Ok((channel, target, data)) => {
+            let json_body = helpers::channel_listing_results_json(&channel, &target, &data);
+            HttpResponse::Ok().header(http::header::CONTENT_TYPE, headers::APPLICATION_JSON)
+                              .header(http::header::CACHE_CONTROL,
+                                      headers::Cache::NoCache.to_string())
+                              .body(json_body)
+        }
+        Err(Error::NotFound) => HttpResponse::new(StatusCode::NOT_FOUND),
+        Err(Error::BadRequest) => HttpResponse::new(StatusCode::BAD_REQUEST),
+        Err(err) => {
+            debug!("Failed to get package, err={}", err);
+            err.into()
+        }
+    }
+}
+
 // Internal - these functions should return Result<..>
 //
 
+fn do_get_latest_channel_packages(req: &HttpRequest,
+                                  qtarget: &Query<Target>,
+                                  origin: &str,
+                                  channel: &ChannelIdent)
+                                  -> Result<(String, String, Vec<BuilderPackageIdent>)> {
+    let opt_session_id = match authorize_session(&req, None, None) {
+        Ok(session) => Some(session.get_id()),
+        Err(_) => None,
+    };
+
+    // This is a new API, so we only look at the query string not the headers.
+    let target = match qtarget.target {
+        Some(ref t) => {
+            trace!("Query requested target = {}", t);
+            t
+        }
+        None => return Err(Error::BadRequest),
```

Comment:
This makes me wonder if defaulting to `x86_64-linux` would be more delightful first time experience when a user discovers this endpoint? I believe often users will intuitively GET an endpoint first without passing arguments in order to interrogate its typical output. If they don't like what they see or want refining, they begin passing in arguments. Just a hunch. Thoughts? 

We should also probably document this route in our `components/builder-api/doc/api.raml` since `target` is a meaningful, if note required, argument.

Reply:
Definitely should add this to RAML. 

So my understanding is that target was added when we added our 2nd platform. Initially everything was x86_64-linux, but when we added windows we had to add the the query parameter to existing APIs. We made it optional to preserve backwards compatibility. I suspect in a future where we add more targets, that default makes less sense. In any event, making it mandatory on a new API seemed reasonable since there was no back-compat issues 
The reasoning behind requiring target is that